tremor-script/lib/std/array.tremor

```diff
@@ -73,3 +73,13 @@ intrinsic fn coalesce(array) as array::coalesce;
 ##
 ## Returns a `string`.
 intrinsic fn join(array, string) as array::join;
+
+## Concatenates two arrays returning a new array. The new array is not a set,
+## i.e. it can contain duplicates depending on the input arrays.
+##
+## ```tremor
+## array::concatenate([1, 2, 3], [3, 4]) => [1, 2, 3, 3, 4]
```

Comment:
I see this is copied from above but perhaps we can change `=>` to `==` instead? (same for the `=` and `=>` above while we spotted it) at one point wed' love to have doctests and that'd be a great start to have it in valid syntax :) (not your fault this was us messing it up in the first place :P)

Reply:
Should we change it to `==` or to `=`? The latter makes more sense imo, as it implies that the executing the `array::concatenate([1, 2, 3], [3, 4])` will result in `[1, 2, 3, 3, 4]`, whereas the former implies an assertion.

Reply:
I'd say lets do it as all the other docs are done. E.g. for the records it is all `==`: https://docs.tremor.rs/tremor-script/stdlib/std/record/

If we want to change it, lets move that to another issue.

Reply:
Ja the assertion is what we'd be looking for, doc tests are example code that gets executed and it's validity tested. This doesn't work yet - it's on the long list of things we want to get done :D - but it's what we're preparing for so `==` is better then `=` which would be illegal since you can't assign to a function execution